plugins/wporg-5ftf/includes/pledge-form.php

```diff
@@ -232,7 +232,26 @@ function process_form_manage( $pledge_id, $auth_token ) {
 
 	PledgeMeta\save_pledge_meta( $pledge_id, $submission );
 
-	// @todo Upload & attach logo.
+	if ( isset( $_FILES['org-logo'], $_FILES['org-logo']['tmp_name'] ) && ! empty( $_FILES['org-logo']['tmp_name'] ) ) {
+		$current_logo_id    = get_post_thumbnail_id( $pledge_id );
+		$logo_attachment_id = upload_image( $_FILES['org-logo'] );
+		if ( is_wp_error( $logo_attachment_id ) ) {
+			return $logo_attachment_id;
+		}
+
+		// Attach new logo to the pledge.
+		wp_update_post( array(
+			'ID'          => $logo_attachment_id,
+			'post_parent' => $pledge_id,
+		) );
+		$updated = set_post_thumbnail( $pledge_id, $logo_attachment_id );
+
+		// Trash the old logo.
+		if ( $updated ) {
+			wp_delete_attachment( $current_logo_id );
```

Comment:
This is kind of an extreme edge case, but it would be possible for a pledge to get an existing image in the media library added to it as a logo, without that image being a child attachment of the pledge post. If that media library image was also used elsewhere, and the pledge were then edited through the front end form, we'd end up with a broken image wherever else it was being used.

This could probably be avoided by just adding a check here to see if the attachment has the right `post_parent` before deleting it.

Reply:
I suppose if someone in wp-admin went in and changed the pledge using the media library to an existing pledge's logo, then yes, but not by the front-end uploading flow.

I can make the change though, since it is technically possible.